tangos/util/read_datasets_file.py

```diff
@@ -5,7 +5,6 @@ def read_datasets(basedir,filename):
     if os.path.exists(os.path.join(basedir, "datasets.txt")):
         with open(os.path.join(basedir, "datasets.txt")) as f:
             for l in f:
-                if l.split()[0].endswith(filename):
+                if filename in l.split()[0]:
                     return int(l.split()[1])
-    else:
-        raise AssertionError("Unable to open datasets.txt")
+    raise AssertionError("Unable to open datasets.txt")
```

Comment:
This now seems like a fall-through so the error message could be updated accordingly (i.e. it might be able to open it but just not find the requested filename)